pkg/synopsysctl/cmd_migrate.go

```diff
@@ -771,9 +771,94 @@ func addNameLabels(namespace string, name string, appName string) error {
 	return nil
 }
 
+// migrateCleanupCmd cleanup the unused resources
+var migrateCleanupCmd = &cobra.Command{
+	Use:           "cleanup",
+	Example:       "synopsysctl migrate cleanup <from>\nsynopsysctl migrate cleanup 2019.4.2\nsynopsysctl migrate cleanup <from> -n <namespace>",
+	Short:         "Cleanup the unused resources that are created by Synopsys Operator after verification",
+	SilenceUsage:  true,
+	SilenceErrors: true,
+	Args: func(cmd *cobra.Command, args []string) error {
+		// Check number of arguments
+		if len(args) != 1 {
+			cmd.Help()
+			return fmt.Errorf("this command takes 1 argument")
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(namespace) > 0 {
+			return cleanup(namespace)
+		}
+
+		// get operator namespace
+		isClusterScoped := util.GetClusterScope(apiExtensionClient)
+		if isClusterScoped {
+			namespaces, err := util.GetOperatorNamespace(kubeClient, metav1.NamespaceAll)
+			if err != nil {
+				return err
+			}
+
+			if len(namespaces) > 1 {
+				return fmt.Errorf("more than 1 Synopsys Operator found in your cluster. please pass the namespace of the Synopsys Operator that you want to migrate")
+			}
+			return cleanup(namespaces[0])
+		}
+
+		log.Errorf("namespace of the Synopsys Operator need to be provided")
+		return nil
+	},
+}
+
+// cleanup will cleanup the resources
+func cleanup(namespace string) error {
+	blackDucks, err := util.ListHubs(blackDuckClient, namespace)
+	if err != nil {
+		return fmt.Errorf("failed to list Black Duck instances in namespace '%s' due to %+v", namespace, err)
+	}
+	for _, blackDuck := range blackDucks.Items {
+		blackDuckName := blackDuck.Name
+		blackDuckNamespace := blackDuck.Spec.Namespace
+		log.Infof("cleaning up the Black Duck '%s' in namespace '%s'...", blackDuckName, blackDuckNamespace)
+		if blackDuck.Spec.PersistentStorage {
+			pods, err := util.ListPodsWithLabels(kubeClient, namespace, "job-name=migrate-upload-cache")
+			if err != nil {
+				return fmt.Errorf("failed to list pods in namespace '%s' due to %+v", namespace, err)
```

Comment:
failed to find the upload-cache migration pod in namespace ...

Reply:
done